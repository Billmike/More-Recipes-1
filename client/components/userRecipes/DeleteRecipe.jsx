import React from 'react';
import PropTypes from 'prop-types';
import Dialog from 'material-ui/Dialog';
import FlatButton from 'material-ui/FlatButton';

const propTypes = {
  recipeId: PropTypes.number,
  open: PropTypes.bool.isRequired,
  handleClose: PropTypes.func.isRequired,
  handleDeleteRecipe: PropTypes.func.isRequired,
  deleteMessage: PropTypes.string.isRequired,
  actionTitle: PropTypes.string.isRequired,
  action: PropTypes.string.isRequired,
};

const defaultProps = {
  recipeId: null,
};

/**
 * Delete recipe component
 *
 * @param {object} props - The properties passed to the function
 *
 * @returns {object} React element
 */
const DeleteRecipe = (props) => {
  const { recipeId, open, handleClose, handleDeleteRecipe,
    deleteMessage, actionTitle, action } = props;

  const actions = [
    <FlatButton
      label="Cancel"
      onClick={handleClose}
    />,
    <FlatButton
      label={action || 'Delete'}
      secondary
      onClick={() => handleDeleteRecipe(recipeId, actionTitle)}
    />,
  ];

  return (
    <div>
      <Dialog
        title={actionTitle}
        actions={actions}
        modal
        open={open}
        autoScrollBodyContent
      >
        <div id="delete-recipe">
          <div className="row">
            <div className="col s2">
              <i className="material-icons icon-red large">warning</i>
            </div>
            <div className="col s10">
              <p>{deleteMessage}</p>
            </div>
          </div>
        </div>
      </Dialog>
    </div>
  );
};

DeleteRecipe.propTypes = propTypes;

DeleteRecipe.defaultProps = defaultProps;

export default DeleteRecipe;
