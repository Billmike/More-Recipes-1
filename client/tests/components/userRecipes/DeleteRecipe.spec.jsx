import React from 'react';

import DeleteRecipe from '../../../components/userRecipes/DeleteRecipe';

describe('DeleteRecipe component', () => {
  it('should render component', () => {
    const props = {
      recipeId: 5,
      open: true,
      handleClose: jest.fn(),
      handleDeleteRecipe: jest.fn(),
      deleteMessage: 'Are you sure you want to delete this recipe',
      actionTitle: 'Delete Recipe',
      action: 'Delete',
      userRecipes: {
        isLoading: false,
      },
    };
    const wrapper = shallow(<DeleteRecipe {...props} />);

    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });
});
