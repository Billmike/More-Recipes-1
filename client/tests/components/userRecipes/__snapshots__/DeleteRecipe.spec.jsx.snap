// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteRecipe component should render component 1`] = `
<div>
  <Dialog
    actions={
      Array [
        <FlatButton
          disabled={false}
          fullWidth={false}
          id="cancel-delete-button"
          label="Cancel"
          labelPosition="after"
          labelStyle={Object {}}
          onClick={[MockFunction]}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          primary={false}
          secondary={false}
        />,
        <FlatButton
          disabled={false}
          fullWidth={false}
          id="confirm-delete-button"
          label="Delete"
          labelPosition="after"
          labelStyle={Object {}}
          onClick={[Function]}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          primary={false}
          secondary={true}
        />,
      ]
    }
    autoDetectWindowHeight={true}
    autoScrollBodyContent={true}
    modal={true}
    open={true}
    repositionOnUpdate={true}
    title={
      <div
        className="modal-title center-align"
      >
        Delete Recipe
      </div>
    }
  >
    <div
      id="delete-recipe"
    >
      <div
        className="row center-align"
      >
        <div
          className="col s12 m6 l2"
        >
          <i
            className="material-icons icon-red large"
          >
            warning
          </i>
        </div>
        <div
          className="col s12 m6 l8"
        >
          <p
            id="delete-message"
          >
            Are you sure you want to delete this recipe
          </p>
        </div>
      </div>
    </div>
  </Dialog>
</div>
`;
